### PR TITLE
chore: update actions to use `google-github-actions/setup-gcloud@v1`

### PR DESCRIPTION
**Pull request by iuioiua**
This PR upgrades `google-github-actions/setup-gcloud@v0` to the recommended `google-github-actions/setup-gcloud@v1` within the `build`, `publish-canary` and `update-dl-version` jobs.

### CODE REVIEW

**Review by bartlomieju — COMMENTED** [review]
Are there no changes required with the version upgrade?

**Comment by iuioiua** [comment]
Your question prompted me to double-check. Now, `google-github-actions/auth@v1` is used before setup. Unfortunately, there's no way to test these changes until a release is made.

**Comment by bartlomieju** [comment]
Note to self: if landed, apply these changes to https://github.com/denoland/deno/pull/14448

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -111,6 +111,20 @@ jobs:
...
+      - name: Authenticate with Google Cloud
+        if: |
+          matrix.profile == 'release' &&
+          matrix.job == 'test' &&
+          github.repository == 'denoland/deno' &&
+          (github.ref == 'refs/heads/main' ||
+          startsWith(github.ref, 'refs/tags/'))
+        uses: google-github-actions/auth@v1
+        with:
+          project_id: denoland
```
Comment (bartlomieju) [inline .github/workflows/ci.yml:123]: Where did you take that project_id from @iuioiua?

Comment (bartlomieju) [reply .github/workflows/ci.yml:123]: Nvm, I see below

**Comment by bartlomieju** [comment]
Thanks, I included your changes in https://github.com/denoland/deno/pull/17358, gonna close this one.

**Pull request closed without merging**